### PR TITLE
LPS-199557 Add POSHI automation | Master

### PR DESCRIPTION
**Pull request by alessandromalves**
This is only for master
https://issues.liferay.com/browse/LPS-199557
cc.: @david-gutierrez-mesa @Tim-Cao

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by alessandromalves** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/categories/Categories.testcase [hunk 1]
```diff
@@ -929,6 +929,52 @@ definition {
 		}
 	}
 
+	@description = "This is a use case for LPS-195345. Verify that the order of the categories remains as previously selected after publishing a web content."
+	@priority = 4
+	test viewCategoriesRemainingInTheOrderAddedOnWebContent {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/categories/Categories.testcase:934]: Move to WebContent.testcase file Then renamed to AddWebContentWithCategoryHiearchy

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/categories/Categories.testcase [hunk 2]
```diff
@@ -929,6 +929,52 @@ definition {
...
+		task ("Given that some categories are added to the vocabulary") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/categories/Categories.testcase:935]: Given the user has a category hiearchy

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/categories/Categories.testcase [hunk 3]
```diff
@@ -929,6 +929,52 @@ definition {
...
+					groupName = "Test Site Name",
+					parentCategoryName = "Category Name ${parentCategoryNumber}",
+					title = "Category Name ${subCategoryNumber}",
+					vocabularyName = "Vocabulary Name");
+
+				var parentCategoryNumber = ${parentCategoryNumber} + 1;
+			}
+		}
+
+		task ("When a new web content is added and the categories are selected") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/categories/Categories.testcase:954]: When the user adds a web content with the category hiearchy

File: portal-web/test/functional/com/liferay/portalweb/macros/journal/WebContent.macro [hunk 4]
```diff
@@ -2020,6 +2020,22 @@ definition {
 		}
 	}
 
+	macro viewCategorizationCategoryByIndex {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/macros/journal/WebContent.macro:2023]: viewCategorizationCategoriesOrderCP

File: portal-web/test/functional/com/liferay/portalweb/macros/journal/WebContent.macro [hunk 5]
```diff
@@ -2020,6 +2020,22 @@ definition {
...
+		Panel.expandPanel(panel = "Categorization");
+
+		var categoryIndex = 1;
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/macros/journal/WebContent.macro:2026]: var index = 1;

File: portal-web/test/functional/com/liferay/portalweb/macros/journal/WebContent.macro [hunk 6]
```diff
@@ -2020,6 +2020,22 @@ definition {
...
+
+		for (var categoryName : list ${categoryNameList}) {
+			AssertTextEquals(
+				key_categoryName = ${categoryName},
+				key_index = ${categoryIndex},
+				locator1 = "AssetCategorization#CATEGORY_BY_INDEX",
+				value1 = ${categoryName});
+
+			var categoryIndex = ${categoryIndex} + 1;
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/macros/journal/WebContent.macro:2035]: var index = ${index} + 1;

File: portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uisections/AssetCategorization.path [hunk 7]
```diff
@@ -20,6 +20,11 @@
 	<td>//div[contains(@id,'assetCategoriesSelector') or contains(@class,'field-content')]//span[contains(.,'${key_categoryName}')]/parent::span</td>
 	<td></td>
 </tr>
+<tr>
+	<td>CATEGORY_BY_INDEX</td>
+	<td>//div[contains(@id,'assetCategoriesSelector') or contains(@class,'field-content')]//span[contains(@class,'label')][${key_index}]//span[contains(.,'${key_categoryName}')]/parent::span</td>
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uisections/AssetCategorization.path:25]: `xpath=(//label[normalize-space()='${key_vocabularyName}']//following-sibling::div//span[contains(@class,'label') and contains(@role,'row')])[${index}][normalize-space()='${key_categoryName}']`

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/categories/Categories.testcase [hunk 8]
```diff
@@ -929,6 +929,52 @@ definition {
...
+			WebContentNavigator.gotoEditCP(webContentTitle = "Web Content Title");
+
+			WebContent.addWithMultipleCategories(
+				categoryNameList = "Category Name 1,Category Name 2,Category Name 3,Category Name 4,Category Name 5,Category Name 6",
+				vocabularyName = "Vocabulary Name",
+				webContentContent = "Web Content Content",
+				webContentTitle = "Web Content Title");
+		}
+
+		task ("Then the categories must remain in the order them were added to the web content") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/categories/Categories.testcase:971]: Then the added categories should keep the same order with hiearchy

**Review by Tim-Cao — CHANGES_REQUESTED** [review]
Hi @alessandromalves, would you mind refactoring according to comments? Thanks

**Comment by alessandromalves** [comment]
ci:test:sf

**Comment by Tim-Cao** [comment]
LGTM!

**Comment by Tim-Cao** [comment]
ci:forward

**Pull request closed without merging**